plugins/python/src/main/kotlin/io/paddle/plugin/python/tasks/generate/AbstractGenerateTask.kt

```diff
@@ -0,0 +1,37 @@
+package io.paddle.plugin.python.tasks.generate
+
+import io.paddle.plugin.python.dependencies.repositories.PyPackageRepository
+import io.paddle.plugin.python.extensions.requirements
+import io.paddle.plugin.python.tasks.PythonPluginTaskGroups
+import io.paddle.project.PaddleProject
+import io.paddle.tasks.Task
+import io.paddle.tasks.incremental.IncrementalTask
+
+abstract class AbstractGenerateTask(project: PaddleProject) : IncrementalTask(project) {
+    override val group: String = PythonPluginTaskGroups.GENERATE
+    override val dependencies: List<Task>
+        get() = listOf(
+            project.tasks.getOrFail("resolveRepositories"),
+            project.tasks.getOrFail("resolveInterpreter"),
+            project.tasks.getOrFail("resolveInterpreter")
+        ) + project.subprojects.getAllTasksById(this.id)
+
+    override fun act() {
+        val notResolved = project.requirements.descriptors
+        val resolved = project.requirements.resolved
+        val parsedDependencies = buildList {
+            notResolved.forEach { descriptor ->
+                val resolvedPackage = resolved.find { it.name == descriptor.name }
+                val resolvePath = resolvedPackage?.repo?.let {
+                    if (it != PyPackageRepository.PYPI_REPOSITORY) " @ ${resolvedPackage.distributionUrl} " else ""
```

Comment:
I'm not sure if that is correct, because repos are specified in different ways for Poetry and Pip (reqs). Can you check the docs, pls?
I mean for Poetry, repositories is a separate section in pyproject.toml (and you also need to pass credentials), while  for pip you can just write everything using flags in requirements.txt

Reply:
Oh, I got it, you just added pyproject.toml generation, but the idea was to add compatibility with Poetry, that's different :)

Reply:
Anyway, you don't need to resolve repo for each dependency in requirements, you can basically specify it this way:
```
--extra-index-url https://packages.jetbrains.team/pypi/p/grazi/grazie-ml/simple
grazie_cloud_tools==0.2.39
grazie_tree==0.8.15

gunicorn==20.0.4
json-logging-py==0.2
```